tests/testthat/test_pipelines.R

```diff
@@ -107,3 +107,69 @@ test_that("survtoregr 3", {
   expect_prediction_surv(p)
   expect_true("distr" %in% p$predict_types)
 })
+
+skip_if_not_installed("mlr3learners")
+
+test_that("survtoclassif", {
+  requireNamespace("mlr3learners")
+
+  pipe = mlr3pipelines::ppl("survtoclassif", learner = lrn("classif.log_reg"))
+  expect_class(pipe, "Graph")
+  grlrn = mlr3pipelines::ppl("survtoclassif", learner = lrn("classif.log_reg"),
+                             graph_learner = TRUE)
+  expect_class(grlrn, "GraphLearner")
+  grlrn$train(task)
+  p = grlrn$predict(task)
+  expect_prediction_surv(p)
+
+  cox = lrn("surv.coxph")
+  suppressWarnings(cox$train(task))
+  p2 = cox$predict(task)
+
+  expect_equal(p$truth, p2$truth)
+  expect_equal(p$score(), p2$score(), tolerance = 0.1)
+
+  # Test with cut
+  grlrn = mlr3pipelines::ppl("survtoclassif", learner = lrn("classif.log_reg"),
+                            cut = c(10, 30, 50), graph_learner = TRUE)
+  expect_class(grlrn, "GraphLearner")
+  suppressWarnings(grlrn$train(task))
+  p = grlrn$predict(task)
+  expect_prediction_surv(p)
+
+  # Test with max event time
+  max_time = task$data()[status == 1, min(time)]
+  grlrn = mlr3pipelines::ppl("survtoclassif", learner = lrn("classif.log_reg"),
+                             max_time = max_time, graph_learner = TRUE)
+  expect_error(grlrn$train(task))
+
+  grlrn = mlr3pipelines::ppl("survtoclassif", learner = lrn("classif.log_reg"),
+                             max_time = max_time + 1, graph_learner = TRUE)
+  suppressWarnings(grlrn$train(task))
+  p = grlrn$predict(task)
+  expect_prediction_surv(p)
+
+  # Test with rhs
+  grlrn = ppl("survtoclassif", learner = lrn("classif.log_reg"), rhs = "1",
+              graph_learner = TRUE)
+  grlrn$train(task)
```

Comment:
train here is rank deficit and predict gives warnings, in the next it is not, I am surprized that the output matricies are the same - maybe not hte best dataset to check for this?

Reply:
eg should it be the same trained model?